apps/examples/testcase/le_tc/network/tc_net_poll.c

```diff
@@ -0,0 +1,183 @@
+#include <lwip/sockets.h>
+#include <tinyara/os_api_test_drv.h>
+#include "tc_internal.h"
+
+/**
+* @testcase             :tc_net_poll_teardown_p
+* @brief                :check for poll teardown
+* @scenario             :
+* @apicovered           :lwip_poll()
+* @precondition         :
+* @postcondition        :
+*/
+
+static void tc_net_poll_teardown_p(int fd, int sd, struct poll_test_args *args)
+{
+	int ret = -1;
+	ret = ioctl(fd, TESTIOC_NET_POLL, args);
+	TC_ASSERT_GEQ_CLEANUP("lwip_poll", ret, 0, close(sd));
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase             :tc_net_poll_p
+* @brief                :check for poll setup
+* @scenario             :
+* @apicovered           :lwip_poll()
+* @precondition         :
+* @postcondition        :
+*/
+
+static void tc_net_poll_p(int fd)
+{
+	int ret = -1;
+	int sd = -1;
+	struct pollfd fds[1];
+	struct poll_test_args args;
+
+	sd = socket(AF_INET, SOCK_DGRAM, 0);
+
+	fds[0].fd = sd;
+	fds[0].events = POLLIN;
+
+	args.fd = sd;
+	args.fds = fds;
+	args.setup = 1;
+
+	ret = ioctl(fd, TESTIOC_NET_POLL, &args);
+	TC_ASSERT_GEQ("lwip_poll", ret, 0);
+	TC_SUCCESS_RESULT();
+
+	args.setup = 0;
+
+	tc_net_poll_teardown_p(fd, sd, &args);
+
+	close(sd);
+}
+
+/**
+* @testcase             :tc_net_poll_fds_n
+* @brief                :test with wrong values of fds.
+* @scenario             :
+* @apicovered           :lwip_poll()
+* @precondition         :
+* @postcondition        :
+*/
+
+static void tc_net_poll_fds_n(int fd)
+{
+	int ret = -1;
+	int sd = -1;
+	struct poll_test_args args;
+
+	sd = socket(AF_INET, SOCK_DGRAM, 0);
+
+	args.fd = sd;
+	args.fds = NULL;
+	args.setup = 1;
+
+	ret = ioctl(fd, TESTIOC_NET_POLL, &args);
+	TC_ASSERT_LEQ_CLEANUP("lwip_poll", ret, -1, close(sd));
+	TC_SUCCESS_RESULT();
+	close(sd);
+}
+
+/**
+* @testcase             :tc_net_poll_events_n
+* @brief                :test with wrong value of events
+* @scenario             :
+* @apicovered           :lwip_poll()
+* @precondition         :
+* @postcondition        :
+*/
+
+static void tc_net_poll_events_n(int fd)
+{
+	int ret = -1;
+	int sd = -1;
+	struct pollfd fds[1];
+	struct poll_test_args args;
+
+	sd = socket(AF_INET, SOCK_DGRAM, 0);
+
+	fds[0].fd = sd;
+	args.fds = 0;
+
+	args.fd = sd;
+	args.fds = fds;
+	args.setup = 1;
+
+	ret = ioctl(fd, TESTIOC_NET_POLL, &args);
+	TC_ASSERT_GEQ_CLEANUP("lwip_poll", ret, 0, close(sd));
+	TC_SUCCESS_RESULT();
+	close(sd);
+}
+
+/**
+* @testcase             :tc_net_poll_sd_n
+* @brief                :test with wrong value of socket descriptor
+* @scenario             :
+* @apicovered           :lwip_poll()
+* @precondition         :
+* @postcondition        :
+*/
+
+static void tc_net_poll_sd_n(int fd)
+{
+	int ret = -1;
+	int sd = -1;
+	struct pollfd fds[1];
+	struct poll_test_args args;
+
+	fds[0].fd = sd;
+	fds[0].events = POLLIN;
+
+	args.fd = -1;
+	args.fds = fds;
+	args.setup = 0;
+
+	ret = ioctl(fd, TESTIOC_NET_POLL, &args);
+	TC_ASSERT_LEQ("lwip_poll", ret, -1);
```

Comment:
Is it okay not to close the fd?

Reply:
we have not created any socket here, so no need to close.